backend-shared/util/operations/types_test.go

```diff
@@ -102,6 +103,188 @@ var _ = Describe("Testing CreateOperation function.", func() {
 			Expect(dbOperationFirst.SeqID).To(Equal(dbOperationSecond.SeqID))
 		})
 	})
+
+	Context("Testing waitForOperationToComplete and IsOperationComplete function.", func() {
+		var ctx context.Context
+		var dbq db.AllDatabaseQueries
+
+		delayUpdateOfState := func(operation *db.Operation, k8sClient client.Client, gitopsEngineInstance db.GitopsEngineInstance) {
+			Eventually(func() bool {
+				// Allow waitForOperationToComplete to iterate/wait
+				time.Sleep(2 * time.Second)
+
+				operationget := db.Operation{
+					Operation_id: operation.Operation_id,
+				}
+
+				err := dbq.GetOperationById(ctx, &operationget)
+				// The operation might not be created yet
+				if err != nil {
+					return false
+				}
+
+				Expect(operationget.State).To(BeIdenticalTo(db.OperationState_Waiting))
+
+				// Update the operation state to Completed
+				operationget.State = db.OperationState_Completed
+				err = dbq.UpdateOperation(ctx, &operationget)
+				Expect(err).ToNot(HaveOccurred())
+
+				// Get the operation again and check state
+				err = dbq.GetOperationById(ctx, &operationget)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(operationget.State).To(BeIdenticalTo(db.OperationState_Completed))
+
+				return true
+			}, "5s", "10ms").Should(BeTrue())
+		}
+
+		AfterEach(func() {
+			defer dbq.CloseDatabase()
```

Comment:
Can you please remove defer here as it is in AfterEach block?

Reply:
Done